libs/MailDB/include/MailDB/IMailDB.h

```diff
@@ -1,9 +1,90 @@
 #pragma once
 
+#include <vector>
+#include <string>
+#include <cstdint>
+
+#include "MailException.h"
+
+namespace ISXMailDB
+{
+
+struct User
+{
+    User(std::string user_name, std::string user_password, std::string host_name)
+        : user_name{std::move(user_name)}, user_password{std::move(user_password)}, host_name{std::move(host_name)}
+    {
+    }
+
+    std::string user_name;
+    std::string user_password;
+    std::string host_name;
+};
+
+struct Mail
+{
+    Mail(const std::string_view recipient, const std::string_view sender,
+         const std::string_view subject, const std::string_view body)
+        : recipient(recipient), sender(sender), 
+          subject(subject), body(body)
+    {
+    }
+    
+    std::string recipient;
+    std::string sender;
+    std::string subject;
+    std::string body;
+};
+
+std::ostream& operator<<(std::ostream& os, const Mail& mail);
+
+
 class IMailDB
 {
 public:
-    
-private:
+    IMailDB(const std::string_view host_name)
+    {
+        if (host_name.empty())
+        {
+            throw MailException("Host name couldn't be empty"); // change
+        }
+        m_host_name = host_name;
+    }
+    virtual ~IMailDB() = default;
+
+    IMailDB(const IMailDB&);
+    IMailDB& operator=(const IMailDB&) = delete;
+
+    IMailDB(IMailDB&&) = delete;
+    IMailDB& operator=(IMailDB&&) = delete;
+
+    virtual void Connect(const std::string &connection_string) = 0;
+    virtual void Disconnect() = 0;
+    virtual bool IsConnected() const = 0;
+
+    virtual void SignUp(const std::string_view user_name, const std::string_view hash_password) = 0;
+    virtual void Login(const std::string_view user_name, const std::string_view hash_password) = 0;
+    virtual std::string GetPasswordHash(const std::string_view user_name) = 0;
```

Comment:
Why do we need API to return actual user password hash? Should we force people to use `Login` instead ? Even hashed password without salt could be used to compromise weak passwords.